### PR TITLE
Add `delete-squashed-branches` command

### LINKED ISSUE #858: git delete squashed branches
I would like to propose a new command for deleting squashed branches, something like the following repo does: https://github.com/not-an-aardvark/git-delete-squashed thank you

### PR DESCRIPTION
**Pull request by allejo**
Uses https://github.com/not-an-aardvark/git-delete-squashed as a basis for deleting "squashed-merged" branches.

Fixes #858﻿

---

I hope I'm not _too_ far off with this PR.

### CODE REVIEW

File: bin/git-delete-squashed-branches [hunk 1]
```diff
@@ -0,0 +1,17 @@
+#!/usr/bin/env bash
+
+targetBranch=$1
+
+if [[ -z $targetBranch ]]; then
+  targetBranch=$(git rev-parse --abbrev-ref HEAD)
+else
+  git checkout -q $targetBranch
```
Comment (spacewander) [inline bin/git-delete-squashed-branches:8]: Better to mention the target branch will be checkout as a side effect.

Comment (allejo) [reply bin/git-delete-squashed-branches:8]: Should this be done by removing the `-q` flag or an explicit `echo`?

Comment (spacewander) [reply bin/git-delete-squashed-branches:8]: removing the -q flag is enough.

Comment (allejo) [reply bin/git-delete-squashed-branches]: Done!

**Review by spacewander — APPROVED** [review]

File: Commands.md [hunk 2]
```diff
@@ -851,6 +852,16 @@ Deleted feature/dashboard (was 923befa).
 ...
 ```
 
+## git delete-squashed-branches
+
+Deletes branches that have been "squashed-merged" into a specified branch
+
+```bash
+$ git delete-squashed-branches main
```
Comment (spacewander) [inline Commands.md:860]: Should also update the doc

Comment (allejo) [reply Commands.md:860]: How does this sound to you?

Comment (spacewander) [reply Commands.md:860]: Should we also need to update the output of `git delete-squashed-branches main`?
Would be better if we use `$ (branch) git ...` to indicate the branch change.

Comment (allejo) [reply Commands.md:860]: Sure! Is this a new convention or are there other commands that do this too?

Comment (allejo) [reply Commands.md:860]: How does this sound to you?

~~~markdown
```bash
$ (feature-branch) git delete-squashed-branches main
Deleted branch dependabot/bundler/kramdown-2.3.1 (was 1d3fb00).
Deleted branch dependabot/bundler/rexml-3.2.5 (was a7e4052).
$ (main) git ...
```
~~~

Comment (allejo) [reply Commands.md]: Alrighty! Amended to my last commit

**Review by spacewander — APPROVED** [review]

**Pull request merged**